test/lib/testMayaUsdDirtyScene.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2021 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unittest
+import mayaUtils
+
+import ufe
+
+import maya.cmds as cmds
+import mayaUsd_createStageWithNewLayer
+import mayaUsd
+
+PROXY_NODE_TYPE = "mayaUsdProxyShapeBase"
+
+def getMayaStage():
+    proxyShapes = cmds.ls(type=PROXY_NODE_TYPE, long=True)
+    shapePath = proxyShapes[0]
+    stage = mayaUsd.lib.GetPrim(shapePath).GetStage()
+    return shapePath, stage
+
+def getCleanMayaStage():
+    """ gets a stage that only has an anon layer """
+    cmds.file(new=True, force=True)
+    mayaUsd_createStageWithNewLayer.createStageWithNewLayer()
```

Comment:
Kinda makes you wish that createStageWithNewLayer() would return the stage...  Out of scope for this pull request, of course, but this would be a compatible change we could make.

Reply:
I like that idea. It wasn't that hard to do and I've cleaned up all the other tests.